### PR TITLE
attempt to use shared scripts for build and remove docker use

### PR DESCRIPTION
**Pull request by mjkkirschner**
### Purpose

* Remove docker use
* For most steps use shared scripts
* publish still relies on a custom script as I'd like to leverage the packaging happening during the build, making a nuspec unnecessary. 

I somehow need to test this without merging this PR - I guess I will send another temp PR to the mirror?

### Declarations

Check these if you believe they are true

- [x] The codebase is in a better state after this PR
- [x] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [x] The level of testing this PR includes is appropriate
- [x] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions) and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### Reviewers

(FILL ME IN) Reviewer 1  (If possible, assign the Reviewer for the PR)

(FILL ME IN, optional) Any additional notes to reviewers or testers.

### FYIs

(FILL ME IN, Optional) Names of anyone else you wish to be notified of

### CODE REVIEW

File: cicdscripts/DownloadCICDResources.ps1
Comment (aparajit-pratap) [inline cicdscripts/DownloadCICDResources.ps1:1]: Maybe an improvement for later but I feel we can also move this script to the common CICDResources repo as it seems to be common to projects using the repo.

File: pipeline.yml [hunk 1]
```diff
@@ -55,4 +51,25 @@ ci_test:
     format: "junit"
     pattern: "*.xml"
 
+soc2:
+  harmony:
+    hidden_email_list: dynamo_ws_access
+    third_party_lib_paths:
+      - "src\\GregClient\\bin\\Release\\net8.0\\"
```
Comment (aparajit-pratap) [inline pipeline.yml:58]: Isn't this `bin\\release\\net8.0\\`?

File: pipeline.yml [hunk 2]
```diff
@@ -55,4 +51,25 @@ ci_test:
...
+deployment:
+  
+  -  
+      type: sign
+      files_to_sign:
+        - "src\\GregClient\\bin\\Release\\net8.0\\greg.dll"
+  -
+      type: customized
+      scripts:
+          - "pwsh.exe -ExecutionPolicy ByPass -File .\\cicdscripts\\scripts\\PublishPackage.ps1 -ReleaseBranches master"
```
Comment (aparajit-pratap) [inline pipeline.yml:69]: The path should be `.\\cicdscripts\\PublishPackage.ps1` if I'm not wrong.

File: cicdscripts/PublishPackage.ps1 [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+#>
+$reBranch = [regex]$regex
+
+if ($env:BRANCH_NAME -match $reBranch) 
+{
+	try {
+
+		#deploy already built package.
+		$assemblyPath = "src\GregClient\bin\release"
+		$nupkgFile = Get-ChildItem $assemblyPath\*.nupkg -Depth 1
```
Comment (aparajit-pratap) [inline cicdscripts/PublishPackage.ps1:31]: How is this nupkg file being generated?

File: pipeline.yml (hunk 1 above)
Comment (mjkkirschner) [reply pipeline.yml:58]: no, in this sln there's only one project output we care about so it's directly in the project's bin folder.

File: cicdscripts/PublishPackage.ps1 (hunk 3 above)
Comment (mjkkirschner) [reply cicdscripts/PublishPackage.ps1:31]: https://github.com/DynamoDS/PackageManagerClient/blob/master/src/GregClient/GregClient.csproj#L20
using this csproj property

File: cicdscripts/DownloadCICDResources.ps1
Comment (mjkkirschner) [reply cicdscripts/DownloadCICDResources.ps1:1]: yeah, I am not sure how we'd do that - this is the script that retrieves the common cicd resources... perhaps we could add something to the CI lib directly?

File: pipeline.yml (hunk 1 above)
Comment (mjkkirschner) [reply pipeline.yml:58]: ah, it seems it was like that before and I accidentally removed the output path property from the csproj, let me refer that change.

**Review by aparajit-pratap — APPROVED** [review]

**Comment by mjkkirschner** [comment]
successful build, scan and test on the internal PR here:
https://git.autodesk.com/Dynamo/PackageManagerClient/pull/6
I will merge this and watch the mirror for successful push to nuget - will send another PR if that fails.

**Pull request merged**